### DISCUSSION TITLE
How to use validation with float value in TorchEstimator?

### DISCUSSION THREAD

**Discussion by MichaelLtv** [horovod/horovod#2789] [Q&A]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.7.1
3. Horovod version: 0.21.2
4. MPI version: 
5. CUDA version:
6. NCCL version:
7. Python version: 3.8.5
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.

How do I provide a float value for the validation param in TorchEstimator? I want to give a float value as a validation split giving the percentage of data to be randomly selected for validation. However, when I do pass a float value, it gives me this error:

`TypeError: Invalid param value given for param "validation". Could not convert <class 'float'> to string type`

I assume this is because it expects a string for a validation column? But then how would I provide a float like it says in estimator.py:

`        validation: Optional validation column name (string) where every row in the column is either 1/True or 0/False,
                    or validation split (float) giving percent of data to be randomly selected for validation.`

This is how I have my TorchEstimator set up:

`            torch_estimator = hvd.TorchEstimator(model=model,
                                                 backend=backend,
                                                 store=store,
                                                 optimizer=optimizer,
                                                 loss=lambda input, target: ce_loss(input, torch.max(target,1)[1].long()),
                                                 input_shapes=[[-1, num_feature_cols]],
                                                 feature_cols=['features'],
                                                 label_cols=['Compound_vec'],
                                                 validation=float(0.3),
                                                 batch_size=params['batch_size'],
                                                 val_batch_size=params['batch_size'],
                                                 epochs=params['epochs'],
                                                 verbose=1)`

**Comment by tgaddair** [comment]
Hmm, looks like that should work. @yuduber, can you take a look?

**Comment by yuduber** [comment]
Sure, I will take a look of this.
